### PR TITLE
Liqun/remove number matching

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**:  replace number matching with relaxed comparison

**Motivation and Context**
Number matching in e2e test is not stable.

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1196,12 +1204,13 @@ def run_training_python_frontend_e2e_tests(cwd):
...
-        run_subprocess([
-            'mpirun', '-n', str(ngpus), '-x', 'NCCL_DEBUG=INFO', sys.executable,
-            bert_pretrain_script], cwd=cwd)
+        # skip long run - it is very difficult to search for breaking commit with this long run test.
+        # # a long run
+        # log.debug('RUN: mpirun -n {} ''-x' 'NCCL_DEBUG=INFO'' {} {}'.format(
+        #     ngpus, sys.executable, bert_pretrain_script))
+        # run_subprocess([
+        #     'mpirun', '-n', str(ngpus), '-x', 'NCCL_DEBUG=INFO', sys.executable,
+        #     bert_pretrain_script], cwd=cwd)
```
Comment (thiagocrepaldi) [inline tools/ci_build/build.py:1213]:
```suggestion
```

File: tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-frontend-test-ci-pipeline.yml [hunk 2]
```diff
@@ -26,7 +26,12 @@ jobs:
     continueOnError: true
     condition: always()
 
-  # insert a python frontend test data preparation step here
+  # TODO: enable this script to avoid the need of having a copy of test data at agent VMs
+  # - script: |
+  #     orttraining/tools/ci_test/download_e2e_test_data.py \
+  #     --azure_blob_url https://onnxruntimetestdata.blob.core.windows.net/training/glue_data.zip
+  #     --target_dir /bert_data/hf_data/
+  #   displayName: 'Download training end-to-end test data'
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-frontend-test-ci-pipeline.yml:34]:
```suggestion
```

File: orttraining/tools/ci_test/download_e2e_test_data.py [hunk 3]
```diff
@@ -20,8 +20,8 @@
 import get_azcopy  # noqa: E402
 
 # update these if the E2E test data changes
-ARCHIVE_BLOB_URL = "https://onnxruntimetestdata.blob.core.windows.net/training/onnxruntime_training_data.zip?snapshot=2020-06-15T23:17:35.8314853Z"
-ARCHIVE_SHA256_DIGEST = "B01C169B6550D1A0A6F1B4E2F34AE2A8714B52DBB70AC04DA85D371F691BDFF9"
+# ARCHIVE_BLOB_URL = "https://onnxruntimetestdata.blob.core.windows.net/training/onnxruntime_training_data.zip?snapshot=2020-06-15T23:17:35.8314853Z"
+# ARCHIVE_SHA256_DIGEST = "B01C169B6550D1A0A6F1B4E2F34AE2A8714B52DBB70AC04DA85D371F691BDFF9"
```
Comment (thiagocrepaldi) [inline orttraining/tools/ci_test/download_e2e_test_data.py:24]:
```suggestion
```

**Review by thiagocrepaldi — APPROVED** [review]

**Review by thiagocrepaldi — DISMISSED** [review]

**Review by thiagocrepaldi — APPROVED** [review]

File: orttraining/tools/ci_test/download_e2e_test_data.py [hunk 4]
```diff
@@ -47,15 +43,18 @@ def _check_file_sha256_digest(path, expected_digest):
 
 def main():
   parser = argparse.ArgumentParser(description="Downloads training end-to-end test data.")
-  parser.add_argument("target_dir", help="The test data destination directory.")
+  parser.add_argument("--azure_blob_url", required=True, help="The test data destination directory.")
```
Comment (edgchen1) [inline orttraining/tools/ci_test/download_e2e_test_data.py:46]: this help text is not accurate. same comment for "archive_sha256_digest". also why do we need to make the URL and digest command line arguments? it makes manually calling this script a bit more difficult. they're still effectively hardcoded in the build definition.

File: tools/ci_build/github/linux/docker/scripts/training/requirements.txt [hunk 5]
```diff
@@ -1,10 +1,10 @@
 --pre
--f https://download.pytorch.org/whl/nightly/cu102/torch_nightly.html
+-f https://download.pytorch.org/whl/torch_stable.html
```
Comment (edgchen1) [inline tools/ci_build/github/linux/docker/scripts/training/requirements.txt:2]: do we still need `--pre`?

**Pull request merged**